course3/homework-string/my_string.cc

```diff
@@ -0,0 +1,164 @@
+#include <iostream>
+#include <cstring>
+#include "my_string.h"
+
+MyString::MyString(){}
+
+MyString::MyString(const char* argString)
+{
+        buffer = std::unique_ptr<char[]>(new char[ std::strlen(argString) + 1]);
+        std::strcpy(buffer.get(), argString);;
+}
+
+MyString::MyString(const MyString& argString)
+{
+        *this = argString;
+}
+
+
+MyString::MyString(MyString&& argString){
+        *this = std::move(argString);
+}
+
+const bool MyString::empty() const
+{
+        if (buffer){
+                return buffer[0] == '\0' || buffer[0] == ' ';
```

Comment:
Added without really thinking it through just before committing. Sleepy-minded logic error.